### PR TITLE
Defined a new broker metric for total query processing time 

### PR DESCRIPTION
**Pull request by navina**
### What does this PR do?
Defined a new broker metric for total query processing time and added metric to broker.yaml

Tags to include: `observability` , `release notes`

### Release Notes:
Includes a new broker timer type metric - `queryTotalTimeMs`

### CODE REVIEW

**Comment by navina** [comment]
Review pls : @npawar

File: pinot-controller/src/main/resources/package-lock.json [hunk 1]
```diff
@@ -12,14 +12,6 @@
         "@babel/highlight": "^7.16.7"
       }
     },
-    "@babel/helper-module-imports": {
```
Comment (npawar) [inline pinot-controller/src/main/resources/package-lock.json:15]: revert this file?

File: pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java [hunk 2]
```diff
@@ -585,6 +586,7 @@ private BrokerResponseNative handleRequest(long requestId, String query, JsonNod
     brokerResponse.setTimeUsedMs(totalTimeMs);
     requestContext.setQueryProcessingTime(totalTimeMs);
     augmentStatistics(requestContext, brokerResponse);
+    _brokerMetrics.setValueOfTableGauge(rawTableName, BrokerGauge.QUERY_TOTAL_TIME_MS, totalTimeMs);
```
Comment (npawar) [inline pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java:589]: should this be a BrokerQueryPhase (in addition to Gauge) so we can get the percentiles, average etc?

**Comment by kishoreg** [comment]
How is this different from the totaltimems that we have today?

File: pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java (hunk 2 above)
Comment (navina) [reply pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java:589]: yeah. I wasn't sure if this falls under the "query phase" metrics as it seemed to be applicable for each stage.  I don't think percentiles/average etc are related to `BrokerQueryPhase`, but rather to the metric type itself ? 
Please correct me if I have misunderstood.  :)

File: pinot-controller/src/main/resources/package-lock.json (hunk 1 above)
Comment (navina) [reply pinot-controller/src/main/resources/package-lock.json:15]: urgh.. i thought I removed this from the commit. will fix it. tks!

**Comment by navina** [comment]
> How is this different from the totaltimems that we have today?

Are you referring to `BrokerQueryPhase.QUERY_EXECUTION` metric? It is currently defined as:
```
_brokerMetrics.addPhaseTiming(rawTableName, BrokerQueryPhase.QUERY_EXECUTION,
        executionEndTimeNs - routingEndTimeNs);
 ```
So, it is not the total query processing time. Total query time should start from the request compilation phase and all the way to the end.

File: pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java (hunk 2 above)
Comment (siddharthteotia) [reply pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java:589]: To confirm - the new metric will include the total  latency observed at broker right from the time of arrival and all the other phases ?

Comment (navina) [reply pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java:589]: @siddharthteotia yes.. that's right. 
@npawar I think we need to use a `Timer` metric type if you need percentiles and average. I can change to use that. `BrokerQueryPhase` just labels the phase name to the metric name. Under the hood, it is a timer metric.

Comment (navina) [reply pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java]: `Timer` has rates and distribution stats. `Gauges` don't. (https://metrics.dropwizard.io/4.2.0/getting-started.html#timers)[https://metrics.dropwizard.io/4.2.0/getting-started.html#timers]

I also verified the metrics locally - it is seen in mbeans and getting published to prometheus.

**Comment by navina** [comment]
@npawar can you merge this PR?

**Pull request merged**